src/edu/csus/ecs/pc2/ui/TeamStatusPane.java

```diff
@@ -321,25 +326,65 @@ private void repopulateGrid(boolean populate) {
             teamLabel.setToolTipText(toolTipText);
             centerPane.add(teamName, teamLabel);
         }
+        
```

Comment:
The following (new) block of code comes after the above for-loop which updates each of the components in the centerPane, but before the call to centerPane.repaint().  It seems like it would be clearer if the centerPane.repaint() call resided just after the for-loop (as it used to), and THEN the new code to update the status labels was added. Updating the status labels has nothing to do with the centerPane -- yet putting the code where it is (between the updates to the centerPane contents and the repainting of the centerPane) sort of implies that it is somehow related.

Reply:
pushed fix.